doc/source/dev_guide/custom_reader.rst

```diff
@@ -476,7 +476,7 @@ needs to implement a few methods:
    in the example below.
 
  - the ``get_area_def`` method, that takes as single argument the
-   :class:`~satpy.dataset.DatasetID` for which we want
+   :class:`~satpy.dataset.DataID` for which we want
```

Comment:
The custom reader guide needs a more significant overhaul after the dataset-id change.  For example, the yaml definition for (some?) readers now have `data_identification_keys` that need to be explained here, and references to dataset ids in the example reader should be removed.

Reply:
I linked to the satpy internals doc, do you think it's enough?

Reply:
Maybe not entirely.  It seems that the `data_identification_keys` are optional, so the `yaml` example in `custom_reader.rst` should still work, if I have understood correctly.  However, the example reader in the documentation contains:

```python
    def get_dataset(self, dataset_id, dataset_info):
        if dataset_id.calibration != 'radiance':
            # TODO: implement calibration to reflectance or brightness temperature
            return
```

which should probably be changed to:

```python
    def get_dataset(self, data_id, dataset_info):
        if data_id["calibration"] != 'radiance':
            # TODO: implement calibration to reflectance or brightness temperature
            return
```

or something similar, as I understand we should now use key access rather than attribute access.

Reply:
i fixed this in a very recent commit. `dataset_id` will remain unchanged for now as it comes from the base class I think.